### DISCUSSION TITLE
LoRaWAN_End_Device_Reference.ino question

### DISCUSSION THREAD

**Discussion by michapr** [jgromes/RadioLib#966] [General]
Hi,

still making "experiments" with RadioLib and saved sessions I made some progress ;)
Anywhere not sure about all, so maybe I'm wrong - then sorry...

In the sample code LoRaWAN_End_Device_Reference.ino I have found this, and I think there maybe a mistake inside?

The "state" in result of node.restore() will be not "0" for success, but will be 0xAA/0xAB (OTAA/ABP), right?  

```
  // on EEPROM-enabled boards, after the device has been activated,
  // the session can be restored without rejoining after device power cycle
  // this is intrinsically done when calling `beginOTAA()` with the same keys
  // or if you 'lost' the keys or don't want them included in your sketch
  // you can call `restore()`
  /*
    Serial.print(F("[LoRaWAN] Resuming previous session ... "));
    state = node.restore();
    if(state == RADIOLIB_ERR_NONE) {
      Serial.println(F("success!"));
    } else {
      Serial.print(F("failed, code "));
      Serial.println(state);
      while(true);
    }
  */
```

**Comment by StevenCellist** [comment]
Almost fully correct - it returns 0x01AA or 0x0AB9. Examples are past due for an upgrade, these will be added. Thanks for the heads-up!
